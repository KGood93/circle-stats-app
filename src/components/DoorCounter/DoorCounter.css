.inputLabel, .countLabel, .noteLabel {
    font-family: 'Roboto', sans-serif;
    color: rgb(12, 70, 94);
}

.activeCount {
    font-family: 'Roboto', sans-serif;
    font-size: 40px;
    color: rgb(12, 70, 94);
}

.doorButtons {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.addButton {
    height: 150px;
    width: 150px;
    border-radius: 75px;
    font-size: 90px;
    font-family: 'Roboto', sans-serif;
    color: rgb(240, 198, 90);
    border: solid 3px rgb(12, 70, 94);
    background-color: rgb(9, 142, 175);
}

.oopsButton {
    width: 75px;
    height: 40px;
    margin-top: 15px;
    background-color: rgb(12, 70, 94);
    border: solid 2px rgb(246, 125, 72);
    font-family: 'Press Start 2P', cursive;
    color: rgb(240, 198, 90);
}

.locationLabel {
    font-family: 'Roboto', sans-serif;
    color: rgb(12, 70, 94);
    margin-bottom: 0px;
}

.location {
    height: 30px;
    width: 275px;
    font-family: 'Roboto', sans-serif;
    font-size: 18px;
    border: solid 2px rgb(12, 70, 94);
}

.noteLabel {
    font-family: 'Roboto', sans-serif;
    color: rgb(12, 70, 94);
    margin-bottom: 0px;
}

.noteContent {
    width: 275px;
    margin-top: 0px;
    font-family: 'Roboto', sans-serif;
    font-size: 18px;
    border: solid 2px rgb(12, 70, 94);
}

.mainButton {
    height: 40px;
    background-color: rgb(12, 70, 94);
    border: solid 2px rgb(246, 125, 72);
    margin-top: 10px;
    margin-bottom: 25px;
    font-family: 'Press Start 2P', cursive;
    color: rgb(240, 198, 90);
}

@media screen and (min-width: 768px) {
    .location {
        width: 350px;
    }

    .noteContent {
        width: 350px;
    }
}