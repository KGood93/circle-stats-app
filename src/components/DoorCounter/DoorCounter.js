import React, {Component} from 'react'
import StatsForm from '../StatsForm/StatsForm'
import {Redirect} from 'react-router-dom'
import config from '../../config'
import './DoorCounter.css'

//Update At_count to attendance once updated in api

class DoorCounter extends Component {
    constructor(props) {
        super(props)
        this.state = {
            clicks: 0,
            show: true,
            location: {
                value: '',
                touched: false
            },
            notes: {
                value: '',
                touched: false
            },
            redirect: false
        }
    }

    updateLocation(meetingLocation) {
        this.setState({location: {value: meetingLocation, touched: true}})
    }

    updateNote(meetingNote) {
        this.setState({notes: {value: meetingNote, touched: true}})
    }

    getMeetId() {
        const {meet_id} = this.props.match.params
        //console.log(meet_id)
        const id = parseInt(meet_id)
        return id
    }

    incrementCount = () => {
        this.setState({clicks: this.state.clicks + 1})
    }



    handleSubmit = event => {
        event.preventDefault()
        //console.log(this.state)
        const {location, clicks, notes} = this.state
        //console.log(location)
        //console.log(clicks)
        const meetingId = this.getMeetId()

        const meetingDetails = {
            meet_id: meetingId,
            location: location.value,
            attendance: clicks,
            notes: notes.value
        }

        //console.log(meetingDetails)

        fetch(`${config.API_ENDPOINT}/stats`, {
            method: 'POST',
            headers: {
                'Content-Type':'application/json'
            },
            body: JSON.stringify(meetingDetails)
        })
        .then(
            this.setState({redirect: true})
        )
        .catch(error => {
            console.log({error})
        })
    }

    render() {
        return (
            <div className="doorCounter">
                <StatsForm className="meeting" onSubmit={this.handleSubmit}>
                    <div className="count">
                        <h2 className="countLabel">Number of Attendees</h2>
                        <h2 className="activeCount">{this.state.clicks}</h2>
                        <div className="doorButtons">
                            <button 
                                type="button" 
                                className="addButton"
                                onClick={this.incrementCount}
                            >
                                +
                            </button>
                        </div>
                    </div>    
                        
                    <div className="newLocation">
                        <h2 className="locationLabel">Location</h2>
                        <input 
                            type="text" 
                            className="location" 
                            onChange={e => this.updateLocation(e.target.value)}
                        />
                    </div>

                    <div className="notes">
                        <h2 className="noteLabel">Notes</h2>
                        <textarea 
                        className="noteContent"
                        onChange={e => this.updateNote(e.target.value)}
                        >
                        </textarea>
                    </div>

                    <div className="next">
                        <button type="submit" className="mainButton">Next</button>
                    </div>
                </StatsForm>
                {this.state.redirect && <Redirect to={`/stats/${this.getMeetId()}`}/>}
            </div>
        )
    }
}

export default DoorCounter