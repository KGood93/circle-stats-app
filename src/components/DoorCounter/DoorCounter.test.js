import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import DoorCounter from './DoorCounter';

describe('DoorCounter Component', () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<DoorCounter />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});