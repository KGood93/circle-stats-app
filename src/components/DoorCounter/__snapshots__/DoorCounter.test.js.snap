// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorCounter Component renders without crashing 1`] = `
<div
  className="doorCounter"
>
  <StatsForm
    className="meeting"
    onSubmit={[Function]}
  >
    <div
      className="count"
    >
      <h2
        className="countLabel"
      >
        Number of Attendees
      </h2>
      <h2
        className="activeCount"
      >
        0
      </h2>
      <div
        className="doorButtons"
      >
        <button
          className="addButton"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="oopsButton"
          onClick={[Function]}
          type="button"
        >
          Oops
        </button>
      </div>
    </div>
    <div
      className="newLocation"
    >
      <h2
        className="locationLabel"
      >
        Location
      </h2>
      <input
        className="location"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="notes"
    >
      <h2
        className="noteLabel"
      >
        Notes
      </h2>
      <textarea
        className="noteContent"
        onChange={[Function]}
      />
    </div>
    <div
      className="next"
    >
      <button
        className="mainButton"
        disabled="Location is Required"
        type="submit"
      >
        Next
      </button>
    </div>
  </StatsForm>
</div>
`;
